### ISSUE TITLE
Wrong type added for selectableRowsComponentProps in #501

### ISSUE THREAD

**Issue by MaxInertia** [jbetancur/react-data-table-component#522]
## Issue Check list
- [x] Agree to the [Code of Conduct](https://github.com/jbetancur/react-data-table-component/blob/master/CODE-OF-CONDUCT.md)
- [x] Read the README
- [x] You are using React 16.8.0+
- [x] You installed `styled-components`
- [x] Include relevant code or preferably a [code sandbox](https://codesandbox.io/embed/react-data-table-sandbox-ccyuu
)

## Describe the bug
The typing for `DataTableProps.selectableRowsComponentProps` is `T`, which is the type of data being displayed by the table. This causes type errors when using code from the example in the README (Using Custom Checkboxes and Indeterminate State).

This type was added in #501 

## To Reproduce
Attempt to use [this example in the README](https://github.com/jbetancur/react-data-table-component#using-custom-checkboxes-and-indeterminate-state)

```
import Checkbox from '@mataerial-ui/core/Checkbox';

...

/*
  In this example, the Material Ui ui lib determines its own indeterminate state via the `indeterminate` property.
  Let's override it using selectableRowsComponentProps`
*/
const selectProps = { indeterminate: isIndeterminate => isIndeterminate };

const MyComponent = () => (
  <DataTable
    title="Arnold Movies"
    columns={columns}
    data={data}
    selectableRows
    selectableRowsComponent={Checkbox} // Pass the function only
    selectableRowsComponentProps={selectProps}
  />
);
```

## Expected behavior
No type error for prop `selectableRowsComponentProps`

## Code Sandbox, Screenshots, or Relevant Code
https://codesandbox.io/s/react-data-table-sandbox-txjx0

![image](https://user-images.githubusercontent.com/12981415/78595646-9ea5ed80-7807-11ea-8017-19cfee85c07a.png)

## Versions (please complete the following information)
![image](https://user-images.githubusercontent.com/12981415/78595407-30612b00-7807-11ea-8dd9-f9335fcb763f.png)
 - RDT: 6.40
 - React (RDT requires 16.8.0+): 16.12.0
 - Styled Components: 5.0.0
 - OS: N/A
 - Browser: N/A

## Additional context

Error shown in the codesandbox:
```
(JSX attribute) IDataTableProps<{ id: number; title: string; year: string; runtime: string; genres: string[]; director: string; actors: string; plot: string; posterUrl: string; }>.selectableRowsComponentProps?: {
    id: number;
    title: string;
    year: string;
    runtime: string;
    genres: string[];
    director: string;
    actors: string;
    plot: string;
    posterUrl: string;
}
Type '{ indeterminate: (isIndeterminate: any) => any; }' is missing the following properties from type '{ id: number; title: string; year: string; runtime: string; genres: string[]; director: string; actors: string; plot: string; posterUrl: string; }': id, title, year, runtime, and 5 more.ts(2740)
index.d.ts(69, 3): The expected type comes from property 'selectableRowsComponentProps' which is declared here on type 'IntrinsicAttributes & IDataTableProps<{ id: number; title: string; year: string; runtime: string; genres: string[]; director: string; actors: string; plot: string; posterUrl: string; }>'
```

**Comment by jbetancur** [comment]
This should probably be   selectableRowsComponentProps?: {}; - will fix

**Comment by jbetancur** [comment]
fixed in 6.5.0
